File: LeetCode/Graph/Euler/go/Leetcode565-array-nesting.go
package sbgo

func arrayNesting(nums []int) int {
	n, ans := len(nums), 0
	for i := 0; i < n; i++ {
		if nums[i] == -1 {
			continue
		}
		ne, cur := i, 0
		for nums[ne] != -1 {
			cur++
			t := ne
			ne = nums[ne]
			nums[t] = -1
		}
		if ans < cur {
			ans = cur
		}
	}
	return ans
}

File: LeetCode/Graph/Euler/Js/Leetcode565-array-nesting.js
var arrayNesting = function (nums) {
    const n = nums.length;
    let ans = 0;
    for (let i = 0; i < n; i++) {
        if (nums[i] == -1) continue;
        let [ne, cur] = [i, 0];
        while (nums[ne] != -1) {
            cur += 1;
            const t = ne;
            ne = nums[ne];
            nums[t] = -1;
        }
        ans = Math.max(ans, cur);
    }
    return ans;
};